File: app/src/main/java/com/holokenmod/GridSize.java
package com.holokenmod;

import androidx.annotation.NonNull;

import org.apache.commons.lang3.StringUtils;

public class GridSize {
	private final int width;
	private final int height;
	
	public GridSize(int width, int height) {
		this.width = width;
		this.height = height;
	}
	
	public static GridSize create(String gridSizeString) {
		if (StringUtils.isNumeric(gridSizeString)) {
			int size = Integer.valueOf(gridSizeString);
			
			return new GridSize(size, size);
		}
		
		String[] parts = StringUtils.split(gridSizeString, "x");
		int width = Integer.valueOf(parts[0]);
		int height = Integer.valueOf(parts[1]);
		
		return new GridSize(width, height);
	}
	
	public int getSurfaceArea() {
		return width * height;
	}
	
	@NonNull
	@Override
	public String toString() {
		return width + "x" + height;
	}
	
	public int getHeight() {
		return height;
	}
	
	public int getWidth() {
		return width;
	}
	
	public int getAmountOfNumbers() {
		return Math.max(width, height);
	}
	
	public boolean isSquare() {
		return width == height;
	}
}

File: holoken-core/src/main/kotlin/com/holokenmod/grid/GridSize.kt
package com.holokenmod.grid

import kotlin.math.max

data class GridSize(
    val width: Int,
    val height: Int
) {
    val surfaceArea: Int
        get() = width * height

    override fun toString(): String {
        return width.toString() + "x" + height
    }

    val amountOfNumbers: Int
        get() = max(width, height)

    val isSquare: Boolean
        get() = width == height

    companion object {
        @JvmStatic
        fun create(gridSizeString: String): GridSize {
            try {
                val size = gridSizeString.toInt()
                return GridSize(size, size)
            } catch (e: NumberFormatException) {
                val parts = gridSizeString.split("x")
                val width = parts[0].toInt()
                val height = parts[1].toInt()
                return GridSize(width, height)
            }
        }
    }
}
